src/components/forms/update-identity-form.tsx

```diff
@@ -20,121 +21,133 @@ import { fetchJson } from "@src/utils/fetch-json";
 import { getIdentityType } from "@src/utils/get-identity-type";
 
 const UpdateIdentityForm: React.FC<{
-  currentIdentity: Identity;
-}> = ({ currentIdentity }) => {
+  data?: { identity: Identity };
+}> = ({ data }) => {
   const [identity, setIdentity] = React.useState<InMemoryTemporaryIdentity>({
     value: "",
-    type: currentIdentity.type,
+    type: data ? data.identity.type : IdentityTypes.EMAIL,
```

Comment:
Not very proud of this one. If you've got any better idea, I would appreciate 🙂  @Fuzznimp

Reply:
Well, except if we add a `UNKNOWN` enum entry, I guess we can't do better :/.